src/main/java/com/yoga/horus/controller/UserController.java

```diff
@@ -21,29 +21,36 @@ public UserController(UserService userService) {
     }
 
     @GetMapping("/{id}")
-    public APIResponse<UserDTO> getById(@PathVariable String id) {
-        UserDTO user= userService.getOne(UUID.fromString(id));
+    public APIResponse<UserDTO> getById(@PathVariable UUID id) {
+        UserDTO user= userService.getOne(id);
         return APIResponse.ok(user);
     }
 
-
-    public APIResponse<UserDTO> getAll() {
-        return null;
+    @GetMapping("/")
+    public APIResponse<List<UserDTO>> getAll() {
+        List<UserDTO> users = userService.getAll();
+        return APIResponse.ok(users);
     }
 
 
     @PostMapping("/")
-    public APIResponse<UserDTO> create(@RequestBody HttpServletRequest request) {
-        return null;
+    public APIResponse<UserDTO> create(@RequestBody User user) {
+        UserDTO newUser = userService.create(user);
+        return APIResponse.ok(newUser);
     }
 
 
-    public APIResponse<UserDTO> update(UUID id, HttpServletRequest request) {
-        return null;
+    @PutMapping("/{id}")
+    public APIResponse<UserDTO> update(@PathVariable UUID id, @RequestBody User user) {
+        UserDTO updatedUser = userService.update(id, user);
+        return APIResponse.ok(updatedUser);
     }
 
 
-    public APIResponse<UserDTO> delete(UUID id) {
-        return null;
+    @DeleteMapping("/{id}")
+    public APIResponse<String> delete(@PathVariable UUID id) {
+        String deletedObject = Constant.Deleted(userService.getOne(id).getClass());
+        userService.delete(id);
+        return APIResponse.ok(Constant.Deleted(deletedObject));
```

Comment:
What if the operation did not succeed?

Reply:
Resolving this conversation, since its object is cleared.

@Mohamedsabry109 if it is not clear please go ahead and ask in discord, we can continue there.